ledger/ledger.go

```diff
@@ -108,7 +109,7 @@ type Ledger struct {
 // genesisInitState.Accounts specify the initial blocks and accounts to use if the
 // database wasn't initialized before.
 func OpenLedger(
-	log logging.Logger, dbPathPrefix string, dbMem bool, genesisInitState ledgercore.InitState, cfg config.Local,
+	log logging.Logger, hotPrefix string, coldPrefix string, dbMem bool, genesisInitState ledgercore.InitState, cfg config.Local,
```

Comment:
there are many-many unnecessary code changes because of the signature change.
And even after the change, these both new params do not define paths completely since there is a hidden `l.cfg.CatchpoointDir` path as well.

Please consider the following:
1. Making OpenLedger a template accepting string or path provider.
2. Make LedgerPathProvider returning hot/cold/catchpoint
Example is here: https://go.dev/play/p/lKCiBUPJ_uK

Benefits:
1. One less argiment
2. Less code changes
3. No hidden paths from cfg.

Reply:
I like this suggestion. Would make the changes much more manageable and passing the same string twice across all of the tests is distracting as well. 

I'm guessing it would be the first use of generics in this repo as well? 😃

Reply:
the first one was #5497 I guess

Reply:
Agreed, I really like this pattern, and am relieved that I can remove the signature changes. Thank you for the idea :)